themes/default/compositions.cjs

```diff
@@ -78,27 +81,35 @@ module.exports = (config, theme) => ({
       content: '""',
     },
     '&:checked + label > span:before': {
+      '--border-color': 'hsl(var(--tw-colors-primary-dark-hs),var(--tw-colors-primary-dark-l))',
+
       animation: `${config('prefix')}${theme('animation.squish')}`,
-      borderColor: theme('colors.primary.dark'),
+      borderColor: `var(--border-color, ${theme('colors.primary.dark')})`,
       borderWidth: '0.5em',
     },
     '&:focus + label > span:before': {
-      outlineColor: colorLib(theme('ringColor.DEFAULT')).fade(0.1).toString(),
+      '--outline-color': 'hsl(var(--tw-colors-focus-default-hs),var(--tw-colors-focus-default-l), 0.1)',
+      outlineColor: `var(--outline-color, ${colorLib(theme('ringColor.DEFAULT')).fade(0.1).toString()})`,
       outlineOffset: '1px',
       outlineWidth: '3px',
     },
   },
   '.checkbox-input_animated_dark': {
+    
     '&:checked + label > span:before': {
-      borderColor: theme('colors.primary.light'),
+      '--border-color': 'hsl(var(--tw-colors-primary-light-hs),var(--tw-colors-primary-light-l))',
+
+      borderColor: `var(--border-color, ${theme('colors.primary.light')})`,
     },
   },
   /**
    * Dialog box theme settings
    */
   '.dialog-box': {
-    backgroundColor: theme('colors.background.light'),
-    border: `1px solid ${theme('colors.secondary.dark')}`,
+    '--bg-color': 'hsl(var(--tw-colors-background-light-hs),var(--tw-colors-background-light-l))',
+    '--border': '1px solid hsl(var(--tw-colors-secondary-dark-hs),var(--tw-colors-secondary-dark-l))',
```

Comment:
I'm a bit worried about mixing static values with custom properties for some border declarations. I was thinking color would have to be declared separately for fallbacks to work. But, in my testing, it seems to fallback to the default if the color values are not declared, even though they are mixed with other valid values like `1px` and `solid`. Can you make sure this works cross-browser? If it does, then I guess I'm okay with it.

Reply:
I was worried about the same thing but I tested it a few times and, like you also noticed, it seems to fallback to the default if no color declared.